frontend/src/views/admin/matching/applicant-view/grid/grid-item/dropdown.tsx

```diff
@@ -35,57 +41,48 @@ export function GridItemDropdown({
     const canBeHidden =
         match.status !== "assigned" && match.status !== "hidden";
 
-    const assignToPosition = React.useCallback(() => {
-        dispatch(
-            upsertMatch({
-                ...baseMatchValues,
-                stagedAssigned: true,
-                stagedHoursAssigned: match.position.hours_per_assignment || 0,
-            })
-        );
-    }, [baseMatchValues, match, dispatch]);
+    const canBeStarred =
+        match.status !== "assigned" && match.status !== "starred";
 
-    const unassignFromPosition = React.useCallback(() => {
-        dispatch(
-            upsertMatch({
-                ...baseMatchValues,
-                stagedAssigned: false,
-                stagedHoursAssigned: 0,
-            })
-        );
-    }, [baseMatchValues, dispatch]);
+    const canBeUnstarred = match.status === "starred";
 
-    const hideFromPosition = React.useCallback(() => {
-        dispatch(upsertMatch({ ...baseMatchValues, hidden: true }));
-    }, [baseMatchValues, dispatch]);
+    const assignApplicant = useAssignApplicantToPosition(
+        baseMatchValues.positionCode,
+        baseMatchValues.utorid,
+        match.position.hours_per_assignment || 0
+    );
+
+    const unassignApplicant = useUnassignApplicantFromPosition(
```

Comment:
The assign and unassign look like they should be grouped and delivered using the same hook. Do you actually mean to toggle the assignment, or do you need these as separate functions?